src/test/java/org/sagebionetworks/template/repo/RepositoryTemplateBuilderImplTest.java

```diff
@@ -332,8 +341,60 @@ public void testBuildAndDeployDev() throws InterruptedException {
 		assertFalse(resources.has("dev101Table1RepositoryDBHighCPUUtilization"));
 		assertFalse(resources.has("dev101Table1RepositoryDBLowFreeStorageSpace"));
+
+		assertTrue(resources.has("dev101RepositoryDB"));
+		JSONObject repoDB = (JSONObject)resources.get("dev101RepositoryDB");
+		JSONObject dbProps = (JSONObject)repoDB.get("Properties");
+		assertFalse(dbProps.has("DBSnapshotIdentifier"));
+		assertTrue(dbProps.has("DBName"));
 	}
 
+	@Test
+	public void testBuildAndDeployDevFromSnapshot() throws InterruptedException {
+		setupValidBeanstalkConfig();
+		List<String> EXPECTED_SUBNETS = Arrays.asList("subnet1", "subnet2", "subnet4");
+		when(mockCloudFormationClient.getOutput(anyString(), anyString())).thenReturn(String.join(",", EXPECTED_SUBNETS));
+		when(mockEc2Client.getAvailableSubnetsForInstanceType(anyString(), any())).thenReturn(EXPECTED_SUBNETS);
+		when(config.getProperty(PROPERTY_KEY_RDS_REPO_SNAPSHOT_IDENTIFIER)).thenReturn("repoSnapshotIdentifier");
+		String[] tableSnaphotIdentifiers = {"table0SnapshotIdentifier", "table1SnapshotIdentifier"};
+		when(config.getComaSeparatedProperty(PROPERTY_KEY_RDS_TABLES_SNAPSHOT_IDENTIFIERS)).thenReturn(tableSnaphotIdentifiers);
+		stack = "dev";
+		configureStack(stack);
+
+		// call under test
+		builder.buildAndDeploy();
+
+		verify(mockCloudFormationClient, times(4)).createOrUpdateStack(requestCaptor.capture());
+		List<CreateOrUpdateStackRequest> list = requestCaptor.getAllValues();
+		CreateOrUpdateStackRequest request = list.get(0);
+		assertEquals("dev-101-shared-resources", request.getStackName());
+		assertEquals(expectedTags, request.getTags());
+		assertEquals(false, request.getEnableTerminationProtection());
+		assertNotNull(request.getParameters());
+		String bodyJSONString = request.getTemplateBody();
+		assertNotNull(bodyJSONString);
+		System.out.println(bodyJSONString);
+		JSONObject templateJson = new JSONObject(bodyJSONString);
+		JSONObject resources = templateJson.getJSONObject("Resources");
+		assertNotNull(resources);
+
+		verify(mockCwlContextProvider).getLogDescriptors(EnvironmentType.REPOSITORY_SERVICES);
+		verify(mockCwlContextProvider).getLogDescriptors(EnvironmentType.REPOSITORY_WORKERS);
+		verify(mockCwlContextProvider).getLogDescriptors(EnvironmentType.PORTAL);
+
+		// dev should not have alarms
+		assertFalse(resources.has("dev101Table1RepositoryDBAlarmSwapUsage"));
+		assertFalse(resources.has("dev101Table1RepositoryDBAlarmSwapUsage"));
+		assertFalse(resources.has("devd101Table1RepositoryDBHighWriteLatency"));
+		assertFalse(resources.has("dev101Table1RepositoryDBHighCPUUtilization"));
+		assertFalse(resources.has("dev101Table1RepositoryDBLowFreeStorageSpace"));
+
+		assertTrue(resources.has("dev101RepositoryDB"));
+		JSONObject repoDB = (JSONObject)resources.get("dev101RepositoryDB");
+		JSONObject dbProps = (JSONObject)repoDB.get("Properties");
+		assertTrue(dbProps.has("DBSnapshotIdentifier"));
+		assertFalse(dbProps.has("DBName"));
```

Comment:
We should probably check the tables identifiers as well and the edge case where there is an inconsistency between the main and tables identifiers "presence"